src/api/autoconnect.ts

```diff
@@ -0,0 +1,70 @@
+import Web3Connect from 'web3connect'
+import { Provider, WalletConnectProvider as WCProvider } from '@gnosis.pm/dapp-ui'
+import WalletConnectProvider from '@walletconnect/web3-provider'
+import { WalletApi } from 'types'
+import { delay } from 'utils'
+import { INFURA_ID } from 'const'
+
+const getWCIfConnected = async (): Promise<WCProvider | null> => {
+  const provider = new WalletConnectProvider({
+    infuraId: INFURA_ID,
+  })
+
+  // TODO: listen for specific websocket events when they are available
+  // hack to wait while WC tries connection with last session's key and uri
+  await delay(1000)
```

Comment:
Good enough :) 
let's create an issue for when https://github.com/WalletConnect/walletconnect-monorepo/pull/209 is merged